Assets/Mirror/Tests/Runtime/NetworkServerTest.cs

```diff
@@ -45,5 +44,46 @@ public IEnumerator DestroyPlayerForConnectionTest()
 
             Assert.That(player == null, "Player should be destroyed with DestroyPlayerForConnection");
         }
+
+        [UnityTest]
+        public IEnumerator DisconnectTimeoutTest()
+        {
+            // message handlers
+            NetworkServer.RegisterHandler<ConnectMessage>((conn, msg) => { }, false);
+            NetworkServer.RegisterHandler<DisconnectMessage>((conn, msg) => { }, false);
+            NetworkServer.RegisterHandler<ErrorMessage>((conn, msg) => { }, false);
+
+            // Set high ping frequency so no NetworkPingMessage is generated
+            NetworkTime.PingFrequency = 20f;
+
+            // listen
+            NetworkServer.Listen(2);
+            Assert.That(NetworkServer.connections.Count, Is.EqualTo(0));
+
+            NetworkClient.ConnectHost();
+            ULocalConnectionToClient localConnection = NetworkServer.localConnection as ULocalConnectionToClient;
+            Assert.That(NetworkServer.localConnection, Is.Not.Null);
+            localConnection.serverIdleTimeout = 5f;
+
+            GameObject RemotePlayer = new GameObject("RemotePlayer", typeof(NetworkIdentity));
+            NetworkConnectionToClient remoteConnection = new NetworkConnectionToClient(1);
+            remoteConnection.serverIdleTimeout = 5f;
+            NetworkServer.OnConnected(remoteConnection);
+            NetworkServer.AddPlayerForConnection(remoteConnection, RemotePlayer);
+
+            Assert.That(NetworkServer.connections.Count, Is.EqualTo(1));
+
+            // wait 10 seconds for remoteConnection to timeout as idle
```

Comment:
can we reduce the timeout to 1 second or less?  having 10 second test is very long

Reply:
Let's get the CI test to pass and then I'll shorten it.

Reply:
I put the timeout to 1 sec and the wait loop to 2 seconds to ensure there's enough time after the connect for it to actually time out and kick.  I don't want a false positive because the test runner lagged.  Even at this, when I run the test locally the kick is practically immediate, and that's kinda scary.